data/campaigns/fri05.wmf/scripting/init_p1.lua

```diff
@@ -0,0 +1,2975 @@
+          wares=             {
+            },
+        },
+      [113] =         {
+          x= 88,
+          y= 188,
+          wares=             {
+            },
+        },
+      [114] =         {
+          x= 91,
+          y= 194,
+          wares=             {
+            },
+        },
+      [115] =         {
+          x= 93,
+          y= 194,
+          wares=             {
+            },
+        },
+      [116] =         {
+          x= 92,
+          y= 197,
+          wares=             {
+            },
+        },
+      [117] =         {
+          x= 89,
+          y= 195,
+          wares=             {
+            },
+        },
+      [118] =         {
+          x= 91,
+          y= 199,
+          wares=             {
+            },
+        },
+      [119] =         {
+          x= 93,
+          y= 194,
+          wares=             {
+            },
+        },
+      [120] =         {
+          x= 90,
+          y= 200,
+          wares=             {
+            },
+        },
+      [121] =         {
+          x= 88,
+          y= 201,
+          wares=             {
+            },
+        },
+      [122] =         {
+          x= 89,
+          y= 133,
+          wares=             {
+            },
+        },
+      [123] =         {
+          x= 80,
+          y= 169,
+          wares=             {
+            },
+        },
+      [124] =         {
+          x= 83,
+          y= 173,
+          wares=             {
+            },
+        },
+    },
+}
+
+function place_flags(eco)
+
+   for i = 1, #eco.flags do
+      local flag = p1:place_flag(map:get_field(eco.flags[i].x, eco.flags[i].y),true)
+      for ware,count in pairs(eco.flags[i].wares) do
+         flag:set_wares(ware,count)
+      end
+   end
+end
+
+function place_roads(eco)
+   for i = 1, #eco.roads do
+      local command = string.format("road = p1:place_road(\"normal\", map:get_field(%i,%i).immovable, %s ,true)", eco.roads[i].x, eco.roads[i].y, eco.roads[i].dirs)
+      local f = assert (load (command))
+      f()
+      -- Second carrier can't be placed currently, so we just fake a carrier
+      worker = (p1.tribe_name.."_carrier")
+      road:set_workers(worker,1)
+   end
+end
+
+function place_buildings(eco)
+
+   for i = 1, #eco.buildings do
+      building = p1:place_building(eco.buildings[i].name, map:get_field(eco.buildings[i].x, eco.buildings[i].y), false, true)
+
+      if eco.buildings[i].inputs then
+         for ware,count in pairs(eco.buildings[i].inputs) do
+            building:set_inputs(ware,count)
+         end
+      end
+
+      if eco.buildings[i].workers then
+         local up = {}
+         for worker,count in pairs(eco.buildings[i].workers) do
+            if count == 0 then
+               up[#up+1] = worker
+            elseif count > 1 and #up > 0 and not building.descr.type_name == "warehouse" then
+               for i = 1, count - 1 do
+                  building:set_workers(up[i],1)
```

Comment:
although there is no occurence as far as I could verify but set_workers is not additive for buildings I believe.
basically these lua functions like set_wares and set_workers and so on behve differntly if used for flags or buildings. I will fix them in atl02 and I am already working on a branch for this.